app/controllers/concerns/sign_in_duration_concern.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module SignInDurationConcern
+  extend ActiveSupport::Concern
+
+  def sign_in_duration_seconds
+    return unless session[:sign_in_page_visited_at]
+    (Time.zone.now - Time.zone.parse(session[:sign_in_page_visited_at])).seconds.to_i
```

Comment:
Non-blocking: I feel like it'd be nice to have a little bit of sub-second precision here, since I'd expect the typical value to be somewhat low where precision matters, especially if this is always rounding down. Alternatively we could `seconds.round`.

Reply:
Yeah I think that's better.

That also brings it a lot closer into alignment with the value for `session_duration` as well, which is nice.